wsd/COOLWSD.cpp

```diff
@@ -990,6 +990,15 @@ class AppConfigMap final : public Poco::Util::MapConfiguration
             setRaw(pair.first, pair.second);
         }
     }
+
+    void updateAppConfigMap(const std::map<std::string, std::string>& map)
+    {
+        clear();
```

Comment:
Nice!

But `update` doesn't imply "clear everything and replace with the new values". Do we really have to clear? Why can't we just add new entries and replace existing ones, without removing anything?

Notice that if you really need to call `clear()` for your case, then it's best to rename this helper, since this can be used in other places. Perhaps `reset`, or just `set`, is a good name.

Also, no need to repeat the class name in the member functions. `update()`, `reset()`, `set()`, are all good names, no need for `AppConfigMap` everywhere.